pkg/util/cmapichecker/cmapichecker.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmapichecker
+
+import (
+	"context"
+
+	errors "github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	// Use v1alpha2 API to ensure that the API server has also connected to the
+	// cert-manager conversion webhook.
+	// TODO(wallrj): Only change this when the old deprecated APIs are removed,
+	// at which point the conversion webhook may be removed anyway.
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+)
```

Comment:
Consider whether if this package had some public Error types it'd be easier to log nice human readable messages from `kubectl cert-manager check api`. E.g.

```
var (
    ErrAPIServerUnreachable = NewError("unable to connect to the Kubernetes API server")
    ErrCertManagerCRDsNotFound = NewError("the cert-manager CRDs are not installed on the Kubernetes API server")
    ErrCertManagerAPIEndpointsNotEstablished = NewError("the cert-manager API endpoints have not yet been published by the Kubernetes API server")
    ErrWebhookConnectionFailure = NewError("the cert-manager webhook server can't be reached.")
)
```